workflow/controller/operator.go

```diff
@@ -1024,6 +1071,21 @@ func (woc *wfOperationCtx) failSuspendedAndPendingNodesAfterDeadlineOrShutdown()
 	}
 }
 
+func (woc *wfOperationCtx) getWorkflowTaskSet() (*wfv1.WorkflowTaskSet, error) {
+	obj, exist, err := woc.controller.wfTaskSetInformer.Informer().GetIndexer().GetByKey(woc.wf.Namespace + "/" + woc.wf.Name)
+	if err != nil {
+		return nil, err
+	}
+	if !exist {
+		return nil, nil
```

Comment:
should this be an error?

Reply:
no, the workflow may or may not have taskset. `taskSetReconciliation` will continue if there is not workflow taskset.